### PR TITLE
Store microtask queues directly in GlobalScope

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18480)

### CODE REVIEW

**Comment by nox** [comment]
@bors-servo try
r? @asajeffrey 
Cc @jdm

File: components/script/dom/dissimilaroriginwindow.rs [hunk 1]
```diff
@@ -63,16 +63,18 @@ impl DissimilarOriginWindow {
                 global_to_clone_from.resource_threads().clone(),
                 timer_event_chan,
                 global_to_clone_from.origin().clone(),
+                // FIXME(nox): The microtask queue is probably not important
+                // here, but this whole DOM interface is a hack anyway.
+                global_to_clone_from.microtask_queue().clone(),
```
Comment (jdm) [inline components/script/dom/dissimilaroriginwindow.rs]: Wrong commit?

**Comment by jdm** [comment]
Clever!

File: components/script/dom/dissimilaroriginwindow.rs (hunk 1 above)
Comment (nox) [reply components/script/dom/dissimilaroriginwindow.rs]: Good catch, fixed!

File: components/script/dom/globalscope.rs [hunk 2]
```diff
@@ -99,6 +100,15 @@ pub struct GlobalScope {
 
     /// The origin of the globalscope
     origin: MutableOrigin,
+
+    /// The microtask queue associated with this global.
+    ///
+    /// It is refcounted because windows in the same script thread shares the
```
Comment (jdm) [inline components/script/dom/globalscope.rs]: s/shares/share/.

**Review by jdm — APPROVED** [review]

File: components/script/dom/globalscope.rs (hunk 2 above)
Comment (nox) [reply components/script/dom/globalscope.rs]: Fixed.

**Pull request merged**